docs/spec.md

```diff
@@ -4184,7 +4184,7 @@ where:
 * `padded_inputs = pad(inputs..., init_values..., padding[:, 0], padding[:, 1],
   base_dilations - 1)`.
 * `window_start = result_index * window_strides`.
-* `window_end = window_start + window_dimensions + window_dilations - 1`.
+* `window_end = window_start + window_dimensions + (window_dimensions - 1) * (window_dilations - 1)`.
```

Comment:
Let's use the formula from C15: `dilated_window_shape = window_dimensions = 0 ? 0 : (window_dimensions - 1) * window_dilations + 1`. It does the same thing, but also works correctly for the case when window_dimensions contains zeros.

Reply:
To conform to SliceOp's `C3: 0 <= start_indices <= limit_indices <= shape(operand)`, let's use:
```
windowEnd = any_of(windowDimensions, isZero) ? windowStart : windowStart + (windowDimensions - 1) * windowDilations + 1;
```

Reply:
Sounds good. This makes sense for the interpreter - since even if a single dimension size is zero, then we don't need to do any computation. For shape inference, we'd need to do that on a per-dimension basis (so C15 is written the way it is).

Reply:
Nvm. SliceOp's `C5: 0 < window_dimensions` doesn't allow `windowDimensions` to be 0 in the first place. Assuming we are using `stablehlo.slice`, this check is redundant. I think `windowEnd = windowStart + (windowDimensions - 1) * windowDilations + 1` is enough.

Reply:
Good point. Let's update C15 as well then to drop the "window_dimensions = 0 ? 0 :" part there as well.

Reply:
Done.